src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -2584,6 +2584,7 @@ because xbuild doesn't support framework reference assemblies.
   <Javac
     JavaPlatformJarPath="$(JavaPlatformJarPath)"
     ClassesOutputDirectory="$(IntermediateOutputPath)android\bin\classes"
+    ClassesZip="$(IntermediateOutputPath)android\bin\classes.zip"
```

Comment:
@garuma I wanted to verify that the designer does not in fact use this file -- and it uses the `.class` files directly? `SetupDependenciesForDesigner` will no longer generate `classes.zip`.

If `classes.zip` is used, I can revisit this and do something else.

Reply:
@jonathanpeppers ideally we should use the classes.zip. We switched to that to get around various command line and path length issues.

Reply:
This won't change a regular build, we will still use `classes.zip` as input to dx/d8.

With this change, the designer won't generate `classes.zip` (if it doesn't use it), so that prevents us from re-running the dx/d8 step on an incremental build.

What I was seeing:
* Edit layout in designer
* Save, designer runs some targets, `classes.zip` changes
* On the next build we run dx/d8 again, even though nothing on the Java side changed.

Reply:
my concern here is that the designer build will fail because of path length issues...

Reply:
The designer build doesn't run dx/d8, right? Unless I missed something?

I think the designer just needs aapt/javac to be able to load.

It seems like `classes.zip` isn't used by the designer at all, because the integration tests passed.

Reply:
> I think the designer just needs aapt/javac to be able to load.

Correct